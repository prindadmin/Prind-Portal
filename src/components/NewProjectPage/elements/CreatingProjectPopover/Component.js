import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
