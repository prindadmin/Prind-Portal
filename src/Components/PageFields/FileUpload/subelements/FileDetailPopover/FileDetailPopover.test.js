
import { shallow } from 'enzyme';
import Component from './FileDetailPopover'

var props;

beforeEach(() => {
  props = {
    chosenFileDetails: {
      uploadName: "File123.txt",
      uploadDateTime: "2021-04-06T07:06:23",
      uploadedBy: "Testy McTesterson",
      proofLink: "https://prind.tech",
      signatures: [
        {
          signerName: "Signer One",
          signatureDateTime: "2021-01-11T11:11:11",
          proofLink: "https://testnet.explorer.factom.pro/entries/93ae94fce1d960dc818c049f9a23f8305cdea54dd6954153c57be62179db1fcf",
        },
        {
          signerName: "Signer Two",
          signatureDateTime: "2021-02-22T22:22:22",
          proofLink: "https://testnet.explorer.factom.pro/entries/93ae94fce1d960dc818c049f9a23f8305cdea54dd6954153c57be62179db1fcf",
        }
      ]
    },
    projects: {
      chosenProject: {
        projectId: "5678"
      }
    },
    pageName: "design",
    fieldID: "1",
    onClosePopover: jest.fn(),
  }
});

it('FileDetailPopover - render', async () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('FileDetailPopover - click file-details-popover-container', async () => {
  const mockStopPropagation = jest.fn()
  const component = shallow(<Component {...props} />);
  const container = component.find('.file-details-popover-container')
  container.simulate('click', { stopPropagation: mockStopPropagation })
  expect(mockStopPropagation).toHaveBeenCalledTimes(1);
});

it('FileDetailPopover - click popup-greyer', async () => {
  const mockStopPropagation = jest.fn()
  const mockClosePopover = jest.fn()
  props.onClosePopover = mockClosePopover
  const component = shallow(<Component {...props} />);
  const container = component.find('#popup-greyer')
  container.simulate('click', { stopPropagation: mockStopPropagation })
  expect(mockStopPropagation).toHaveBeenCalledTimes(1);
  expect(mockClosePopover).toHaveBeenCalledTimes(1);
});

it('CurrentVersion - Should click proof link div and stop propagation', () => {
  const mockStopPropagation = jest.fn()
  const component = shallow(<Component {...props} />);
  const link = component.find('#proof-link-container')
  link.simulate('click', { stopPropagation: mockStopPropagation })
  expect(mockStopPropagation).toHaveBeenCalledTimes(1)
});
