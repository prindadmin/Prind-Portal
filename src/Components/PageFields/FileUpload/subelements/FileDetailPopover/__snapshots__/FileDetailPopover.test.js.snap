// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDetailPopover - render 1`] = `
<PopOverHandler>
  <div
    id="popup-greyer"
    onClick={[Function]}
  >
    <div
      id="file-details-popover"
    >
      <div
        id="popup-box"
      >
        <div
          className="file-details-popover-container"
          onClick={[Function]}
        >
          <div
            className="element-title"
          >
            File Details
          </div>
          <div
            className="element-description"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="file-details"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="field-names col-4"
                    >
                      <div>
                        File Name: 
                      </div>
                      <div>
                        Uploaded At: 
                      </div>
                      <div>
                        Uploaded By: 
                      </div>
                      <div>
                        Proof: 
                      </div>
                    </div>
                    <div
                      className="field-values col-auto"
                    >
                      <div>
                        File123.txt
                      </div>
                      <div>
                        <br />
                      </div>
                      <div>
                        Testy McTesterson
                      </div>
                      <div
                        id="proof-link-container"
                        onClick={[Function]}
                      >
                        <a
                          href="undefined/entries/prind.tech"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Link to Proof
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col download-box-cell"
              >
                <withRouter(Connect(DownloadBox))
                  fieldID="1"
                  fileVersionDetails={
                    Object {
                      "proofLink": "https://prind.tech",
                      "signatures": Array [
                        Object {
                          "proofLink": "https://testnet.explorer.factom.pro/entries/93ae94fce1d960dc818c049f9a23f8305cdea54dd6954153c57be62179db1fcf",
                          "signatureDateTime": "2021-01-11T11:11:11",
                          "signerName": "Signer One",
                        },
                        Object {
                          "proofLink": "https://testnet.explorer.factom.pro/entries/93ae94fce1d960dc818c049f9a23f8305cdea54dd6954153c57be62179db1fcf",
                          "signatureDateTime": "2021-02-22T22:22:22",
                          "signerName": "Signer Two",
                        },
                      ],
                      "uploadDateTime": "2021-04-06T07:06:23",
                      "uploadName": "File123.txt",
                      "uploadedBy": "Testy McTesterson",
                    }
                  }
                  onDownloadFailure={[Function]}
                  onDownloadSuccess={[Function]}
                  pageName="design"
                  projectId="5678"
                />
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="signature-details"
                >
                  <div>
                    <SignatureHistory
                      details={
                        Array [
                          Object {
                            "proofLink": "https://testnet.explorer.factom.pro/entries/93ae94fce1d960dc818c049f9a23f8305cdea54dd6954153c57be62179db1fcf",
                            "signatureDateTime": "2021-01-11T11:11:11",
                            "signerName": "Signer One",
                          },
                          Object {
                            "proofLink": "https://testnet.explorer.factom.pro/entries/93ae94fce1d960dc818c049f9a23f8305cdea54dd6954153c57be62179db1fcf",
                            "signatureDateTime": "2021-02-22T22:22:22",
                            "signerName": "Signer Two",
                          },
                        ]
                      }
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</PopOverHandler>
`;
